packages/buckets/src/api/index.ts

```diff
@@ -403,10 +405,13 @@ export async function bucketsPushPath(
       client.send(req)
 
       if (source.content) {
-        for await (const chunk of source.content) {
-          const part = new PushPathRequest()
-          part.setChunk(chunk as Buffer)
-          client.send(part)
+        const process = await block({ size: 4096, noPad: true })
+        for await (const chunk of process(source.content)) {
+          for (const buf of chunk._bufs) {
+            const part = new PushPathRequest()
+            part.setChunk(buf as Buffer)
+            client.send(part)
+          }
```

Comment:
This makes me slightly nervous, as its not really how we should consume the bl object. Since its already doing the chucking for us, what about (https://www.npmjs.com/package/bl#blslice-start--end--):

```suggestion
          // New Buffer object containing bytes of chunk, even if split over multiple sub-buffers
          const buf = chunk.slice()
          const part = new PushPathRequest()
          part.setChunk(buf as Buffer)
          client.send(part)
```